src/csharp/lowlevel-generator/model/model-class.ts

```diff
@@ -237,7 +237,9 @@ export class ModelClass extends Class implements EnhancedTypeDeclaration {
         *body() {
           for (const hp of headerProperties) {
             const hparam = <ModelProperty>hp;
-            yield hparam.assignPrivate(hparam.deserializeFromContainerMember(KnownMediaType.Header, headers, hparam.serializedName));
+            const values = `__${camelCase(['header', ...deconstruct(hparam.serializedName)])}Values`;
+            const tmp = `__${camelCase(['header', ...deconstruct(hparam.serializedName)])}`;
+            yield If(`${valueOf(headers)}.TryGetValues("${hparam.serializedName}", out var ${values})`, `${hparam.backingName} = System.Linq.Enumerable.FirstOrDefault(${values}) is string ${tmp} ? ${tmp} : (string)${hparam.name})`);
```

Comment:
Am I missing a case where it's possible for header values to not be strings?  If they cannot be strings, there's no point in doing the is string, and I think you can just do ..FirstOrDefault(${values}) ?? ${hparam.name}

Reply:
Oh hmmm you're probably right, I was trying to make a few changes as possible.  I can update this in my new round of fixes.